File: app/src/main/java/com/lumostech/remotecontrol/MyApp.java
package com.lumostech.remotecontrol;

import android.app.Application;
import android.content.Context;
import android.provider.Settings;

public class MyApp extends Application {
    public static final String TAG="MyApp";
    private Context context;

    @Override
    public void onCreate() {
        super.onCreate();
        context=this;
        try {
            setMyServiceEnable();
        }catch (Exception ignored){

        }

    }

    /**
     * 需要授予权限 android.permission.WRITE_SECURE_SETTINGS
     * adb shell pm grant 包名 android.permission.WRITE_SECURE_SETTINGS
     */
    private void setMyServiceEnable() {
        String name = getPackageName()+"/"+ MyService.class.getName();

        String string = Settings.Secure.getString(getContentResolver(),
                Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES);
        StringBuffer stringBuffer=new StringBuffer(string);
        if (!string.contains(name)){
            String s = stringBuffer.append(":").append(name).toString();
            Settings.Secure.putString(getContentResolver(),
                    Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES,s);
        }
    }
}

File: app/src/main/java/com/lumostech/remotecontrol/MyApp.kt
package com.lumostech.remotecontrol

import android.app.Application
import android.content.Context
import android.provider.Settings

class MyApp : Application() {
    private var context: Context? = null

    override fun onCreate() {
        super.onCreate()
        context = this
        try {
            setMyServiceEnable()
        } catch (ignored: Exception) {
        }
    }

    /**
     * 需要授予权限 android.permission.WRITE_SECURE_SETTINGS
     * adb shell pm grant 包名 android.permission.WRITE_SECURE_SETTINGS
     */
    private fun setMyServiceEnable() {
        val name = packageName + "/" + MyService::class.java.name

        val string = Settings.Secure.getString(
            contentResolver,
            Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES
        )
        val stringBuffer = StringBuffer(string)
        if (!string.contains(name)) {
            val s = stringBuffer.append(":").append(name).toString()
            Settings.Secure.putString(
                contentResolver,
                Settings.Secure.ENABLED_ACCESSIBILITY_SERVICES, s
            )
        }
    }

    companion object {
        const val TAG: String = "MyApp"
    }
}
